src/Integration/Language.cs

```diff
@@ -127,7 +136,7 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return this.ServerKey.GetHashCode();
+            return this.Id.GetHashCode();
```

Comment:
Another requirement is that GetHashCode is stable. This is as `this.Id` is readonly.